### PR TITLE
 PHOENIX-3036 Modify phoenix IT tests to extend BaseHBaseManagedTimeT…

### PR DESCRIPTION
**Pull request by prakul**
…ableReuseIT

### CODE REVIEW

File: phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java [hunk 1]
```diff
@@ -2063,5 +2080,35 @@ protected static void populateMultiCFTestTable(String tableName, Date date) thro
...
+
+        ResultSet rs = phxConn.createStatement().executeQuery(ddl);
+        //boolean res =
+        while(rs.next()){
+            String ten = rs.getString("TENANT_ID");
+            String seqN = rs.getString("SEQUENCE_SCHEMA");
+            String seqaN = rs.getString("SEQUENCE_NAME");
+            String seqNam = rs.getString("SEQUENCE_SCHEMA");
+        }
+        /*if(exists) {
```
Comment (samarthjain) [inline phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java]: This shouldn't be commented out, I think.

File: phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java [hunk 2]
```diff
@@ -2063,5 +2080,35 @@ protected static void populateMultiCFTestTable(String tableName, Date date) thro
...
+                + PhoenixDatabaseMetaData.SEQUENCE_NAME
+                + " FROM " + PhoenixDatabaseMetaData.SYSTEM_SEQUENCE
+                + " WHERE ";
+
+        ddl += " TENANT_ID  " + ((tenantID == null ) ? "IS NULL " : " = '" + tenantID + "'");
+        ddl += " AND SEQUENCE_NAME " + ((sequenceName == null) ? "IS NULL " : " = '" +  sequenceName + "'");
+        ddl += " AND SEQUENCE_SCHEMA " + ((sequenceSchemaName == null) ? "IS NULL " : " = '" + sequenceSchemaName + "'" );
+
+        ResultSet rs = phxConn.createStatement().executeQuery(ddl);
+        //boolean res =
```
Comment (samarthjain) [inline phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java]: Remove commented code.

File: pom.xml [hunk 3]
```diff
@@ -226,6 +226,7 @@
           <artifactId>maven-failsafe-plugin</artifactId>
           <version>${maven-failsafe-plugin.version}</version>
           <executions>
+
```
Comment (samarthjain) [inline pom.xml]: Revert changes to this file since it has only white space changes.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/AppendOnlySchemaIT.java [hunk 4]
```diff
@@ -274,36 +285,15 @@ public void testTableAddColumnsDifferentClient() throws Exception {
         testAddColumns(false);
     }
 
-    public void testCreateTableDropColumns() throws Exception {
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/AppendOnlySchemaIT.java]: Why was this test case removed?

File: phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java [hunk 5]
```diff
@@ -1052,16 +1057,20 @@ private static void deletePriorSequences(long ts, Connection globalConn) throws
...
     }
-    
+
     protected static void initSumDoubleValues(byte[][] splits, String url) throws Exception {
-        ensureTableCreated(url, "SumDoubleTest", splits);
+        initSumDoubleValues("SumDoubleTest", splits, url);
+    }
+
+    protected static void initSumDoubleValues(String tableName, byte[][] splits, String url) throws Exception {
+        ensureTableCreated(url, tableName, "SumDoubleTest", splits);
```
Comment (samarthjain) [inline phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java]: Minor nit: Define a static constant for SumDoubleTest like other table ddl types.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseHBaseManagedTimeTableReuseIT.java [hunk 6]
```diff
@@ -47,6 +48,10 @@
 @NotThreadSafe
 @Category(HBaseManagedTimeTableReuseTest.class)
 public class BaseHBaseManagedTimeTableReuseIT extends BaseTest {
+    protected static Configuration getTestClusterConfig() {
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseHBaseManagedTimeTableReuseIT.java]: I don't see this method being used anywhere. Please remove if it isn't needed.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java [hunk 7]
```diff
@@ -90,7 +90,8 @@ public BaseQueryIT(String indexDDL) {
     @Before
     public void initTable() throws Exception {
          ts = nextTimestamp();
-        initATableValues(tenantId, getDefaultSplits(tenantId), date=new Date(System.currentTimeMillis()), ts);
+
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId), date=new Date(System.currentTimeMillis()), ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java]: Is there a need to pass "aTable" here?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java [hunk 8]
```diff
@@ -61,7 +61,7 @@
     protected static final String tenantId = getOrganizationId();
     protected static final String ATABLE_INDEX_NAME = "ATABLE_IDX";
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java]: Shouldn't the index name also be a random string?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CSVCommonsLoaderIT.java [hunk 9]
```diff
@@ -45,15 +45,15 @@
...
+public class CSVCommonsLoaderIT extends BaseHBaseManagedTimeTableReuseIT {
 
     private static final String DATATYPE_TABLE = "DATATYPE";
     private static final String DATATYPES_CSV_VALUES = "CKEY, CVARCHAR, CCHAR, CINTEGER, CDECIMAL, CUNSIGNED_INT, CBOOLEAN, CBIGINT, CUNSIGNED_LONG, CTIME, CDATE\n"
             + "KEY1,A,A,2147483647,1.1,0,TRUE,9223372036854775807,0,1990-12-31 10:59:59,1999-12-31 23:59:59\n"
             + "KEY2,B,B,-2147483648,-1.1,2147483647,FALSE,-9223372036854775808,9223372036854775807,2000-01-01 00:00:01,2012-02-29 23:59:59\n"
             + "KEY3,,,,,,,,,,\n";
-    private static final String STOCK_TABLE = "STOCK_SYMBOL";
-    private static final String STOCK_TABLE_MULTI = "STOCK_SYMBOL_MULTI";
+    //private static final String stockTableName = "STOCK_SYMBOL";
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/CSVCommonsLoaderIT.java]: remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java [hunk 10]
```diff
@@ -18,20 +18,7 @@
 package org.apache.phoenix.end2end;
 
 import static org.apache.phoenix.query.QueryConstants.MILLIS_IN_DAY;
-import static org.apache.phoenix.util.TestUtil.ATABLE_NAME;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java]: The import preferences don't look correct. For guidelines see the code conventions section here: http://phoenix.apache.org/contributing.html

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java [hunk 11]
```diff
@@ -57,12 +44,13 @@
...
 
 
-public class DateTimeIT extends BaseHBaseManagedTimeIT {
+public class DateTimeIT extends BaseHBaseManagedTimeTableReuseIT {
 
     protected Connection conn;
     protected Date date;
     protected static final String tenantId = getOrganizationId();
     protected final static String ROW10 = "00D123122312312";
+    protected  String tableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java]: Don't have tableName as a member variable. Instead each test case can create their own table. This will be critical once we decide on parallelizing test methods within a test class.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java [hunk 12]
```diff
@@ -72,7 +72,7 @@ public DerivedTableIT(String[] indexDDL, String[] plans) {
     @Before
     public void initTable() throws Exception {
          ts = nextTimestamp();
-        initATableValues(tenantId, getDefaultSplits(tenantId), null, ts);
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId), null, ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DerivedTableIT.java]: Don't think we need to pass "aTable" here.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/DynamicColumnIT.java [hunk 13]
```diff
@@ -59,16 +59,18 @@
  */
 
 
-public class DynamicColumnIT extends BaseHBaseManagedTimeIT {
+public class DynamicColumnIT extends BaseHBaseManagedTimeTableReuseIT {
     private static final byte[] FAMILY_NAME_A = Bytes.toBytes(SchemaUtil.normalizeIdentifier("A"));
     private static final byte[] FAMILY_NAME_B = Bytes.toBytes(SchemaUtil.normalizeIdentifier("B"));
 
+    private String tableName = "TESTTBL";
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/DynamicColumnIT.java]: Shouldn't the name be generated by generateRandomString() ?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ExpFunctionEnd2EndIT.java [hunk 14]
```diff
@@ -34,21 +34,26 @@
 /**
  * End to end tests for {@link ExpFunction}
  */
-public class ExpFunctionEnd2EndIT extends BaseHBaseManagedTimeIT {
+public class ExpFunctionEnd2EndIT extends BaseHBaseManagedTimeTableReuseIT {
 
     private static final String KEY = "key";
     private static final double ZERO = 1e-8;
+    private String signedTableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ExpFunctionEnd2EndIT.java]: Same comment as before. Don't have table names as member variables as it makes running tests in parallel difficult.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ExtendedQueryExecIT.java [hunk 15]
```diff
@@ -51,7 +51,7 @@ public void testToDateFunctionBind() throws Exception {
         Date date = new Date(1);
         String tenantId = getOrganizationId();
 
-        initATableValues(tenantId, getDefaultSplits(tenantId),date, ts);
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId),date, ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ExtendedQueryExecIT.java]: If you need to pass the atable as name then use the static constant defined in BaseTest.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/HashJoinMoreIT.java [hunk 16]
```diff
@@ -43,7 +43,7 @@
 
 import com.google.common.collect.Maps;
 
-public class HashJoinMoreIT extends BaseHBaseManagedTimeIT {
+public class HashJoinMoreIT extends BaseHBaseManagedTimeTableReuseIT {
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/HashJoinMoreIT.java]: It doesn't seem like this test was converted to use random table names. Please revert the changes to this file.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/LnLogFunctionEnd2EndIT.java [hunk 17]
```diff
@@ -33,10 +33,12 @@
 /**
  * End to end tests for {@link LnFunction} and {@link LogFunction}
  */
-public class LnLogFunctionEnd2EndIT extends BaseHBaseManagedTimeIT {
+public class LnLogFunctionEnd2EndIT extends BaseHBaseManagedTimeTableReuseIT {
 
     private static final String KEY = "key";
     private static final double ZERO = 1e-9;
+    private String signedTableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/LnLogFunctionEnd2EndIT.java]: No member variables for storing table names, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRuntimeIT.java [hunk 18]
```diff
@@ -95,53 +96,62 @@ private static Filter getUserTableAndViewsFilter() {
     }
     
     private void testGetTenantIdExpression(boolean isSalted) throws Exception {
+        //Have to metaData tables because BaseHBaseManagedTimeTableReuseIT doesn't delete them after each test case , and tenant list will create issues between test cases
+        deletePriorMetaData(HConstants.LATEST_TIMESTAMP, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRuntimeIT.java]: Do we need this if we randomly generate the tenantIds?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryDatabaseMetaDataIT.java [hunk 19]
```diff
@@ -1066,7 +1065,7 @@ public void testAddPKColumn() throws Exception {
     public void testDropKVColumn() throws Exception {
         long ts = nextTimestamp();
         String tenantId = getOrganizationId();
-        initATableValues(tenantId, getDefaultSplits(tenantId), null, ts);
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId), null, ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryDatabaseMetaDataIT.java]: Use static constant for "aTable" in BaseTest.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryDatabaseMetaDataIT.java [hunk 20]
```diff
@@ -1106,7 +1105,7 @@ public void testDropKVColumn() throws Exception {
     public void testDropPKColumn() throws Exception {
         long ts = nextTimestamp();
         String tenantId = getOrganizationId();
-        initATableValues(tenantId, getDefaultSplits(tenantId), null, ts);
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId), null, ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryDatabaseMetaDataIT.java]: Use static constant for "aTable" in BaseTest.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java (hunk 8 above)
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java]: Nevermind my comment since this class hasn't been converted yet.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryIT.java [hunk 21]
```diff
@@ -148,7 +148,7 @@ public void testToDateOnString() throws Exception { // TODO: test more conversio
         }
     }
 
-    
+
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryIT.java]: Please revert changes to this file since they only look whitespace related.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/RegexpReplaceFunctionIT.java [hunk 22]
```diff
@@ -33,13 +33,15 @@
 import org.junit.Test;
 
 
-public class RegexpReplaceFunctionIT extends BaseHBaseManagedTimeIT {
+public class RegexpReplaceFunctionIT extends BaseHBaseManagedTimeTableReuseIT {
 
     private int id;
+    protected  String tableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/RegexpReplaceFunctionIT.java]: No member variables for table names, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/RegexpSubstrFunctionIT.java [hunk 23]
```diff
@@ -33,13 +33,15 @@
 import org.junit.Test;
 
 
-public class RegexpSubstrFunctionIT extends BaseHBaseManagedTimeIT {
+public class RegexpSubstrFunctionIT extends BaseHBaseManagedTimeTableReuseIT {
 
     private int id;
+    String tableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/RegexpSubstrFunctionIT.java]: No member variables for table names, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ReverseScanIT.java [hunk 24]
```diff
@@ -52,11 +52,11 @@
 import com.google.common.collect.Maps;
 
 
-public class ReverseScanIT extends BaseHBaseManagedTimeIT {
-    protected static final String ATABLE_INDEX_NAME = "ATABLE_IDX";
+public class ReverseScanIT extends BaseHBaseManagedTimeTableReuseIT {
+    protected static final String ATABLE_INDEX_NAME = generateRandomString();
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ReverseScanIT.java]: The only test case that uses the index is testReverseScanIndex. Please move this there.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ReverseScanIT.java [hunk 25]
```diff
@@ -52,11 +52,11 @@
...
 
     @BeforeClass
-    @Shadower(classBeingShadowed = BaseHBaseManagedTimeIT.class)
+    @Shadower(classBeingShadowed = BaseHBaseManagedTimeTableReuseIT.class)
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ReverseScanIT.java]: I think you can remove the doSetup() method altogether here. We don't need to disable and delete ATABLE_NAME any more.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/RowValueConstructorIT.java [hunk 26]
```diff
@@ -65,7 +65,7 @@
     public void testRowValueConstructorInWhereWithEqualsExpression() throws Exception {
         long ts = nextTimestamp();
         String tenantId = getOrganizationId();
-        initATableValues(tenantId, getDefaultSplits(tenantId), null, ts);
+        initATableValues("aTable", tenantId, getDefaultSplits(tenantId), null, ts, getUrl());
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/RowValueConstructorIT.java]: Static constant, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SpooledTmpFileDeleteIT.java [hunk 27]
```diff
@@ -38,11 +38,12 @@
 
 
 
-public class SpooledTmpFileDeleteIT extends BaseHBaseManagedTimeIT {
+public class SpooledTmpFileDeleteIT extends BaseHBaseManagedTimeTableReuseIT {
 	
     private Connection conn = null;
     private Properties props = null;
     private File spoolDir;
+	private String tableName;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SpooledTmpFileDeleteIT.java]: No table names in member variables, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/AppendOnlySchemaIT.java (hunk 4 above)
Comment (prakul) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/AppendOnlySchemaIT.java]: its a method which was'nt being used.

File: phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java (hunk 5 above)
Comment (prakul) [inline phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java]: Static DDL for SumDoubleTest is already there.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseHBaseManagedTimeTableReuseIT.java (hunk 6 above)
Comment (prakul) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseHBaseManagedTimeTableReuseIT.java]: Its being used in QueryServerBasicsIT, BasePigIT.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java (hunk 7 above)
Comment (prakul) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/BaseQueryIT.java]: First param is tableName. For this class we want tableName to be "aTable" hence passing it directly.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/HashJoinMoreIT.java (hunk 16 above)
Comment (prakul) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/HashJoinMoreIT.java]: This class uses unique name for each test and thus can be derived from BaseHBaseManagedTimeTableReuseIT

**Comment by JamesRTaylor** [comment]
Is this ready to go, @prakul & @samarthjain? Sounds like there are just a few minor issues turned up by Samarth. Let's get this committed to 4.x ASAP, please.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanQueryIT.java [hunk 28]
```diff
@@ -111,10 +113,10 @@ private void initSelectAfterUpsertTable(Connection conn) throws Exception {
     
     @Test
     public void testSkipScanFilterQuery() throws Exception {
-        String createTableDDL = "CREATE TABLE test" + "(col1 VARCHAR," + "col2 VARCHAR," + "col3 VARCHAR,"
+        String createTableDDL = "CREATE TABLE test1" + "(col1 VARCHAR," + "col2 VARCHAR," + "col3 VARCHAR,"
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanQueryIT.java]: Is there a reason why table name here wasn't generated randomly?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanQueryIT.java [hunk 29]
```diff
@@ -246,15 +248,15 @@ public void testVarCharXIntInQuery() throws Exception {
     public void testPreSplitCompositeFixedKey() throws Exception {
         Connection conn = DriverManager.getConnection(getUrl());
         try {
-            conn.createStatement().execute("create table test(key_1 char(3) not null, key_2 char(4) not null, v varchar(8)  CONSTRAINT pk PRIMARY KEY (key_1,key_2)) split on('000','100','200')");
+            conn.createStatement().execute("create table test2(key_1 char(3) not null, key_2 char(4) not null, v varchar(8)  CONSTRAINT pk PRIMARY KEY (key_1,key_2)) split on('000','100','200')");
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanQueryIT.java]: Is there a reason why table name here wasn't generated randomly?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SortMergeJoinMoreIT.java [hunk 30]
```diff
@@ -39,10 +39,10 @@
 
 import com.google.common.collect.Maps;
 
-public class SortMergeJoinMoreIT extends BaseHBaseManagedTimeIT {
+public class SortMergeJoinMoreIT extends BaseHBaseManagedTimeTableReuseIT {
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SortMergeJoinMoreIT.java]: It doesn't look like all test methods were converted to use generateRandomString() for table names.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/TenantIdTypeIT.java [hunk 31]
```diff
@@ -74,7 +74,8 @@ public TenantIdTypeIT(String dataType, String tenantId, String otherTenantId) {
         this.dataType = dataType;
         this.tenantId = tenantId;
         this.otherTenantId = otherTenantId;
-        String tbl = "foo" + dataType;
+        String tbl = generateRandomString();
+        //String tbl = "foo" + dataType;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/TenantIdTypeIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/UpsertBigValuesIT.java [hunk 32]
```diff
@@ -62,7 +62,7 @@ public void testIntegerPK() throws Exception {
         assertTrue(rs.next());
         assertEquals(testNumbers.length, rs.getInt(1));
         assertFalse(rs.next());
-        select = "SELECT pk FROM PKIntValueTest where pk >= " + Integer.MIN_VALUE + 
+        select = "SELECT pk FROM PKIntValueTest where pk >= " + Integer.MIN_VALUE +
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/UpsertBigValuesIT.java]: Table name is not a random string.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/DropMetadataIT.java [hunk 33]
```diff
@@ -84,14 +77,17 @@ public static void doSetup() throws Exception {
...
     public void testDropViewKeepsHTable() throws Exception {
         HBaseAdmin admin = driver.getConnectionQueryServices(getUrl(), TEST_PROPERTIES).getAdmin();
+        String hbaseNativeViewName = generateRandomString();
+
+        byte[] hbaseNativeBytes = SchemaUtil.getTableNameAsBytes(HBASE_NATIVE_SCHEMA_NAME, hbaseNativeViewName);
         try {
             try {
-                admin.disableTable(HBASE_NATIVE_BYTES);
-                admin.deleteTable(HBASE_NATIVE_BYTES);
+                admin.disableTable(hbaseNativeBytes);
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/DropMetadataIT.java]: No need to disable and delete table since you the name is random.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java [hunk 34]
```diff
@@ -307,16 +309,21 @@ public void testDeleteIndexedExpressionMutableLocalIndex() throws Exception {
...
 
     protected void helpTestDeleteIndexedExpression(boolean mutable, boolean localIndex) throws Exception {
-        String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
+        String dataTableName = generateRandomString();
         String fullDataTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + dataTableName;
-        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + "IDX";
+        String indexName = generateRandomString();
+        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + indexName;
+
+        //String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java [hunk 35]
```diff
@@ -380,16 +387,20 @@ public void testDeleteCoveredColMutableLocalIndex() throws Exception {
     }
 
     protected void helpTestDeleteCoveredCol(boolean mutable, boolean localIndex) throws Exception {
-        String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
+        String dataTableName = generateRandomString();
         String fullDataTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + dataTableName;
-        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + "IDX";
+        String indexName = generateRandomString();
+        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + indexName;
+        //String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java [hunk 36]
```diff
@@ -461,14 +472,19 @@ public void testGroupByCountMutableLocalIndex() throws Exception {
     }
 
     protected void helpTestGroupByCount(boolean mutable, boolean localIndex) throws Exception {
-        String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
+        String dataTableName = generateRandomString();
         String fullDataTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + dataTableName;
+        String indexName = generateRandomString();
+        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + indexName;
+
+       // String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java [hunk 37]
```diff
@@ -515,22 +531,27 @@ public void testSelectDistinctMutableLocalIndex() throws Exception {
     }
 
     protected void helpTestSelectDistinct(boolean mutable, boolean localIndex) throws Exception {
-        String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
+        String dataTableName = generateRandomString();
         String fullDataTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + dataTableName;
+        String indexName = generateRandomString();
+        String fullIndexTableName = INDEX_DATA_SCHEMA + QueryConstants.NAME_SEPARATOR + indexName;
+
+       // String dataTableName = mutable ? MUTABLE_INDEX_DATA_TABLE : INDEX_DATA_TABLE;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexMetadataIT.java [hunk 38]
```diff
@@ -354,27 +355,29 @@ public void testAlterIndexWithLowerCaseName() throws Exception {
         Connection conn = DriverManager.getConnection(getUrl(), props);
         conn.setAutoCommit(false);
         String indexName = "\"lowerCaseIndex\"";
+        String indexDataTable = generateRandomString();
+        //String indexName = generateRandomString();
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexMetadataIT.java]: Remove commented code.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/LocalIndexIT.java [hunk 39]
```diff
@@ -76,24 +76,18 @@
...
-    private String indexName = "I";
-    private String indexPhysicalTableName;
-    private TableName physicalTableName;
+    String schemaName="TEST";
 
     public LocalIndexIT(boolean isNamespaceMapped) {
         this.isNamespaceMapped = isNamespaceMapped;
-        this.physicalTableName = SchemaUtil.getPhysicalTableName(tableName.getBytes(), isNamespaceMapped);
-        this.indexPhysicalTableName = this.physicalTableName.getNameAsString();
+        //this.physicalTableName = SchemaUtil.getPhysicalTableName(tableName.getBytes(), isNamespaceMapped);
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/LocalIndexIT.java]: Remove commented code.

File: phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java [hunk 40]
```diff
@@ -23,6 +23,7 @@
 import static org.apache.phoenix.monitoring.GlobalClientMetrics.GLOBAL_FAILED_QUERY_COUNTER;
 import static org.apache.phoenix.monitoring.GlobalClientMetrics.GLOBAL_QUERY_TIMEOUT_COUNTER;
 import static org.apache.phoenix.util.ByteUtil.EMPTY_BYTE_ARRAY;
+import static org.apache.phoenix.util.ByteUtil.getKeyRange;
```
Comment (samarthjain) [inline phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java]: Please revert changes to this file as they are only import related.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexMetadataIT.java [hunk 41]
```diff
@@ -482,20 +482,22 @@ public void testBinaryNonnullableIndex() throws Exception {
             conn.close();
         }
     }
-    
+
     @Test
     public void testAsyncCreatedDate() throws Exception {
+        //Have to delete metaData tables because BaseHBaseManagedTimeTableReuseIT doesn't delete them after each test case , and tenant list will create issues between test cases
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexMetadataIT.java]: Remove the comment and the following line.

**Comment by samarthjain** [comment]
This looks great now @prakul. Did you also make the changes for the test classes where there are member variables for storing table names? The best way to handle those would be to declare a method annotated with @Before which just sets the tableName = generateRandomString(). Are you getting a clean run when doing mvn clean install?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanAfterManualSplitIT.java [hunk 42]
```diff
@@ -43,7 +43,7 @@
 import com.google.common.collect.Maps;
 
 
-public class SkipScanAfterManualSplitIT extends BaseHBaseManagedTimeIT {
+public class SkipScanAfterManualSplitIT extends BaseHBaseManagedTimeTableReuseIT {
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SkipScanAfterManualSplitIT.java]: I think table names can be randomized here. I would also get rid of tableName and tableNameBytes static constants.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/SubqueryUsingSortMergeJoinIT.java [hunk 43]
```diff
@@ -66,16 +66,16 @@
 
 @RunWith(Parameterized.class)
 public class SubqueryUsingSortMergeJoinIT extends BaseHBaseManagedTimeIT {
-    
+
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/SubqueryUsingSortMergeJoinIT.java]: Nothing seems to have changed in this file. Please revert it.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java [hunk 44]
```diff
@@ -10,10 +10,7 @@
 package org.apache.phoenix.end2end.index;
 
 import static org.apache.phoenix.query.QueryConstants.MILLIS_IN_DAY;
-import static org.apache.phoenix.util.TestUtil.INDEX_DATA_SCHEMA;
```
Comment (samarthjain) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexExpressionIT.java]: Re-organize the imports here.
